src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -1063,7 +1063,7 @@ QString QgsPostgresConn::quotedJsonValue( const QVariant &value )
 {
   if ( value.isNull() || !value.isValid() )
     return QStringLiteral( "null" );
-  const auto j { QgsJsonUtils::jsonFromVariant( value ) };
+  const auto j = QgsJsonUtils::jsonFromVariant( value );
```

Comment:
Interesting, not sure what it changes. @elpaso, any idea?

Reply:
Same happens with code changes, bit strange I agree.

Reply:
Apparently `{ QgsJsonUtils::jsonFromVariant( value ) }` returns an array containing one element and that's an object while `= QgsJsonUtils::jsonFromVariant( value )` contains the object itself not nested in an array.

Possibly it's a change between C++ 11 and C++ 17: with 11 `auto initA{1};` is `std::initializer_list<int>` and with 17 it's `int`

After quick tests it seems to solve #30558

Reply:
It looks wierd to me too, but if you checked that with debugger I trust your interpretation of the events. It may also be compiler specific.

Reply:
Is there any advantage? I'd also still go for `foo bar = 1;` or `foo bar(1);`  instead of `foo bar { 1 };`.  But we have a lot of the latter now…

Reply:
Yes, there is an advantage: https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives

The problem is in the last sentence in the accepted answer: it is not always easy to understand when `{}` initialization does not do what you want like it seems in this case.

Reply:
Those answers don't look like there's a common agreement on best practice usage of list initialization. And that there's a difference between standard versions doesn't help to convince either.

Even looking at the accepted (controversial!) answer, it says

> The only situation where = is preferred over {} is when using auto keyword to get the type determined by the initializer.

which if I see it correctly is exactly what's done here.

At the moment I'm like @jef-n a bit concerned that there's a extensive use of new constructs which look a lot like they make life easier but have some hidden caveats which are hard to discover and side-effects which even change between versions of the standard.

Can we come up with proper guidelines to use `auto` and `{}` (proper as in: our average developer is able to follow them without headaches and our average reviewer is able to check and if necessary enforce them within the context of what's visible in a pull request on github diff)?